### PR TITLE
Add List Merging/Routing/SampleData to Flask Templates

### LINKED ISSUE #732: Create Flask Composition Templates to work with List Routing
We need to create templates for List data and routes for a generated flask application to be able to render List route.

### PR DESCRIPTION
**Pull request by streetcoder123**
Task: https://microsoftgarage.visualstudio.com/Intern%20GitHub/_workitems/edit/28593

**Changes**
- Adds List merging to Flask Templates (Adds constants for list endpoints)
- Adds Sample Data to work with List
- Adds Routing to get list items, post a new list item and delete list items

closes #732

### CODE REVIEW

**Review by Tanya0609 — DISMISSED** [review]

**Review by annielovesjs — DISMISSED** [review]

File: templates/Web/_composition/Flask/Page.Flask.List.AddRoutes/server/server_postaction.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+@app.route(CONSTANTS['ENDPOINT']['LIST'])
+def getList():
+    return jsonify(
+        sampleData['listTextAssets']['listItems']
+    )
+
+@app.route(CONSTANTS['ENDPOINT']['LIST'], methods = ['POST'])
+def addListItem():
+    data = request.data.decode('utf-8')
+    data = ast.literal_eval(data)
```
Comment (piotrmark) [inline templates/Web/_composition/Flask/Page.Flask.List.AddRoutes/server/server_postaction.py]: Not a Flask expert here, but this seems over complicated (explicit decoding + ast).
I believe that client posts this data as a form.
Could you try `request.form` instead?
https://stackoverflow.com/questions/10434599/how-to-get-data-received-in-flask-request

File: templates/Web/_composition/Flask/Page.Flask.AddSampleData.List/server/sampleData$wts.Page.React.List_gpostaction.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+//{[{
+shortLoremIpsum = 'Lorem id sint aliqua tempor tempor sit. Ad dolor dolor ut nulla mollit dolore non eiusmod Lorem tempor nisi cillum.';
+//}]}
+sampleData = {}
+
+# This class holds sample data used by some generated pages to show how they can be used.
+# TODO Web Template Studio: Delete this file once your app is using real data.
+//{[{
+sampleData['listTextAssets'] = {
+    'id': 3,
```
Comment (piotrmark) [inline templates/Web/_composition/Flask/Page.Flask.AddSampleData.List/server/sampleData$wts.Page.React.List_gpostaction.py]: I find this name (id) confusing. IMO nextId would be more clear (or listId to be consistent with Node).

File: templates/Web/_composition/Flask/Page.Flask.List.AddRoutes/server/server_postaction.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    sampleData['listTextAssets']['id'] += 1
+    return make_response(
+        jsonify(
+            listItem
+        ), 201
+    )
+
+@app.route(CONSTANTS['ENDPOINT']['LIST'] + '/<int:id>', methods=['DELETE'])
+def deleteListItem(id):
+    listItemToRemove = [listItem for listItem in sampleData['listTextAssets']['listItems'] if listItem['_id'] == id]
```
Comment (piotrmark) [inline templates/Web/_composition/Flask/Page.Flask.List.AddRoutes/server/server_postaction.py]: I think `listItemsToRemove` would be more appropriate here as it returns an array, even though it's expected to be 1-element array.
Following checks will also gain readability: `((len(listItemsToRemove) > 1))`

File: templates/Web/_composition/Flask/Page.Flask.List.AddRoutes/server/server_postaction.py (hunk 1 above)
Comment (streetcoder123) [reply templates/Web/_composition/Flask/Page.Flask.List.AddRoutes/server/server_postaction.py]: `request.form` is particularly used when we get form data from the front-end (example directly from HTML form). Here, we are getting `json` data and hence `request.form` would be empty.
However, I agree that this is complicated. Actually, I have found a much easier way to get json data in Flask. It is using `request.get_json()`.
I have incorporated that change. Thanks for pointing that out :)

**Review by piotrmark — APPROVED** [review]

**Review by Tanya0609 — APPROVED** [review]

**Review by annielovesjs — APPROVED** [review]

**Pull request merged**